packages/kbn-config/src/deprecation/types.ts

```diff
@@ -103,6 +103,18 @@ export interface ConfigDeprecationFactory {
    * ```
    */
   renameFromRoot(oldKey: string, newKey: string, silent?: boolean): ConfigDeprecation;
+  /**
+   * Copy a configuration property from the root configuration to another.
+   *
+   * @example
+   * Copy 'originPlugin.key' to 'destinationplugin.key'
+   * ```typescript
+   * const provider: ConfigDeprecationProvider = ({ copyFromRoot }) => [
+   *   copyFromRoot('originPlugin.key', 'destinationplugin.key'),
+   * ]
+   * ```
+   */
+  copyFromRoot(originKey: string, destinationKey: string): ConfigDeprecation;
```

Comment:
I apologize if I wasn't clear during the initial slack discussion. The implementation is great, however as this is used only as a workaround, and as I can't think of any real deprecation usage where we would need to copy a property to another place while keeping the original key/value, I would really prefer for this to not be exposed on the public core API and be declared as an external `ConfigDeprecation` that lives in the `lens` plugin instead:

```ts
import { copyFromRoot } from './deprecations';

export const config: PluginConfigDescriptor<ConfigSchema> = {
  schema: configSchema,
  deprecations: () => [copyFromRoot('xpack.lens.enabled', 'lens_oss.enabled')],
};
```

That would reduce the risk of additional 'bad' usages of the config deprecation APIs.

Reply:
After syncing with Pierre we decided to keep it on the kbn-config package but not export it from the `ConfigDeprecationFactory`. So I removed it from there and export it from the main entrypoint instead.